data/index.js

```diff
@@ -72,7 +72,19 @@ export function registerReducer( reducerKey, reducer ) {
 	}
 	const store = createStore( reducer, flowRight( enhancers ) );
 	stores[ reducerKey ] = store;
-	store.subscribe( globalListener );
+
+	// Customize subscribe behavior to call listeners only on effective change,
+	// not on every dispatch.
+	let lastState = store.getState();
+	store.subscribe( () => {
+		const state = store.getState();
+		const hasChanged = state !== lastState;
```

Comment:
Per curiosity, do we have a lot of use-cases where a dispatch is triggered without any change in the state?

Reply:
Many of the actions which incur side-effects often don't do anything to state on their own:

- `SETUP_EDITOR`
- `INSERT_DEFAULT_BLOCK`
- `AUTOSAVE`
- `FETCH_REUSABLE_BLOCKS`

Reply:
For some of these, it could make sense to move the effect to be part of the action creator itself. We might not need a standalone `INSERT_DEFAULT_BLOCK` action type if the `insertDefaultBlock` action creator just returned the result of `insertBlock`.